modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java

```diff
@@ -0,0 +1,409 @@
+                return buf.getChar();
+
+            case FLOAT:
+                return buf.getFloat();
+
+            case DOUBLE:
+                return buf.getDouble();
+
+            case STRING:
+                return new String(bytes, 1, bytes.length - 1, StandardCharsets.UTF_8);
+
+            case BOOLEAN | ARRAY: {
+                boolean[] booleans = new boolean[bytes.length - 1];
+
+                for (int i = 0; i < booleans.length; i++) {
+                    booleans[i] = buf.get() == 1;
+                }
+
+                return booleans;
+            }
+
+            case BYTE | ARRAY:
+                return Arrays.copyOfRange(bytes, 1, bytes.length);
+
+            case SHORT | ARRAY: {
+                short[] shorts = new short[bytes.length / 2];
+
+                for (int i = 0; i < shorts.length; i++) {
+                    shorts[i] = buf.getShort();
+                }
+
+                return shorts;
+            }
+
+            case INT | ARRAY: {
+                int[] ints = new int[bytes.length / 4];
+
+                for (int i = 0; i < ints.length; i++) {
+                    ints[i] = buf.getInt();
+                }
+
+                return ints;
+            }
+
+            case LONG | ARRAY: {
+                long[] longs = new long[bytes.length / 8];
+
+                for (int i = 0; i < longs.length; i++) {
+                    longs[i] = buf.getLong();
+                }
+
+                return longs;
+            }
+
+            case CHAR | ARRAY: {
+                char[] chars = new char[bytes.length / 2];
+
+                for (int i = 0; i < chars.length; i++) {
+                    chars[i] = buf.getChar();
+                }
+
+                return chars;
+            }
+
+            case FLOAT | ARRAY: {
+                float[] floats = new float[bytes.length / 4];
+
+                for (int i = 0; i < floats.length; i++) {
+                    floats[i] = buf.getFloat();
+                }
+
+                return floats;
+            }
+
+            case DOUBLE | ARRAY: {
+                double[] doubles = new double[bytes.length / 8];
+
+                for (int i = 0; i < doubles.length; i++) {
+                    doubles[i] = buf.getDouble();
+                }
+
+                return doubles;
+            }
+
+            case STRING | ARRAY: {
+                List<String> res = new ArrayList<>();
+
+                int offset = 1;
+
+                while (offset != bytes.length) {
+                    int size = buf.getInt(offset);
+
+                    res.add(new String(bytes, offset + 4, size));
+
+                    offset += 4 + size;
+                }
+
+                return res.toArray(EMPTY_STRING_ARRAY);
+            }
+
+            default:
+                throw new IllegalArgumentException(Arrays.toString(bytes));
+        }
+    }
+
+    private static ByteBuffer allocateBuffer(int size) {
+        ByteBuffer buf = ByteBuffer.allocate(size);
```

Comment:
These methods can be chained, code will be a little bit shorter

Reply:
Done